### ISSUE TITLE
Slow "CREATE TABLE IF NOT EXISTS.." on existing table

### ISSUE THREAD

**Issue by dforsber** [duckdb/duckdb#1891]
**What does happen?**

```
D CREATE TABLE t AS SELECT * FROM parquet_scan('file.parquet') LIMIT 1000000;
D .timer on
D CREATE TABLE IF NOT EXISTS t AS SELECT * FROM parquet_scan('file.parquet') LIMIT 1000000;
Run Time: real 10.242 user 9.318672 sys 0.953575
D CREATE TABLE t AS SELECT * FROM parquet_scan('file.parquet') LIMIT 1000000;
Run Time: real 0.347 user 0.235668 sys 0.111898
Error: Catalog Error: Table with name "t" already exists!
D CREATE TABLE IF NOT EXISTS t AS SELECT * FROM parquet_scan('file.parquet') LIMIT 1000000;
Run Time: real 10.916 user 9.724671 sys 1.108568
D 
```

**What should happen?**

The command should return immediately as the table already exists.

**To Reproduce**

See above.

**Environment (please complete the following information):**
 - OS: OSX
 - DuckDB Version: 0.2.7 (osx client)

**Before submitting**
- [x] Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?
- [x] Have you tried this on the latest `master` branch? In case you cannot compile, you may find some binaries here: https://github.com/duckdb/duckdb/releases/tag/master-builds

**Comment by Mytherin** [comment]
Thanks for the report! I can confirm there is no early out here and the entire query will be executed before aborting the query. This should be fixed in #1895.

**Comment by Mytherin** [comment]
This should be fixed in the master now.
